src/tiny_sqlite.nim

```diff
@@ -361,17 +363,12 @@ template transaction*(db: DbConn, body: untyped) =
         body
     else:
         db.exec("BEGIN")
-        var ok = true
+        var ok = false
         try:
-            try:
-                body
-            except Exception:
-                ok = false
-                db.exec("ROLLBACK")
-                raise
+            body
+            ok = true
         finally:
-            if ok:
-                db.exec("COMMIT")
+            db.exec(if ok: "COMMIT" else: "ROLLBACK")
```

Comment:
This change is why the tests are failing. The `body` is allowed to contain a `return`, `break`, or similar so `ok = true` might not execute. That the old code is no longer valid seems like an oversight in Nims exception tracking since a bare `raise` statement cannot raise anything that couldn't already be raised by the code in the `try` block :/

Reply:
You are right! I reverted this change, but had to add more stuff so that compiler would not mark every execMany/execScript call as possibly raising an `Exception`. Now the tests are passing.

Reply:
But then I thought if we could understand in the `finally` block whether an exception is being raised, then we would not have to mess with except/raise; and so it seems, getCurrentException() can be used. So the last version is much simpler.